File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.Arrays;
class Solution {
    public String solution(int[] numbers) {
        boolean zero = Arrays.stream(numbers).allMatch(n -> n == 0);
        if (zero) {
            return "0";
        }
        String[] arrStr = Arrays.stream(numbers)
                .mapToObj(String::valueOf)
                .toArray(String[]::new);
        Arrays.sort(arrStr, (a, b) -> (b + a).compareTo(a + b));

        return String.join("",arrStr);
    }
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
from functools import cmp_to_key

def solution(numbers):
    zero = all(n == 0 for n in numbers)
    if zero:
        return "0"

    def custom_compare(a, b):
        return (int(b + a) - int(a + b))
    
    arr_str = sorted(map(str, numbers), key=cmp_to_key(custom_compare))
    result = ''.join(arr_str)

    return result